synth_data_module/formatting_helpers.py

```diff
@@ -0,0 +1,151 @@
+import glob
+import os
+import math
+from abc import ABC, abstractmethod
+from dateutil.relativedelta import relativedelta
+import pandas as pd
+
+
+class Formatter(ABC):
+    @abstractmethod
+    def gather_data(self) -> pd.DataFrame:
+        pass
+
+    @abstractmethod
+    def format_data(self) -> str:
+        pass
+
+    @abstractmethod
+    def suggested_filename(self) -> str:
+        pass
+
+    @abstractmethod
+    def write_data(self, data):
+        pass
+
+
+# Functions for various formatting operations
+def get_procedure_list():
+    # For Procedures, we are not given a principal vs other distinction from Synthea.  We use the first one in the list
+    # as the principal procedure.
+    procedure_list = []
+    for i in range(1, 26):
+        if i == 1:
+            code = {'fieldid': 'proc_p', 'name': f'Principal Procedure Code', 'length': 7, 'justification': 'left'}
+            procedure_list.append(code)
+            date = {'fieldid': 'proc_pdt', 'name': f'Principal Procedure Date', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+            date = {'fieldid': 'proc_pdy', 'name': f'Principal Procedure Days', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+        else:
+            code = {'fieldid': f'oproc{i}', 'name': f'Procedure Code {i}', 'length': 7, 'justification': 'left'}
+            procedure_list.append(code)
+            date = {'fieldid': f'procdt{i}', 'name': f'Procedure Date {i}', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+            date = {'fieldid': f'procdy{i}', 'name': f'Procedure Days {i}', 'length': 8, 'justification': 'left'}
+            procedure_list.append(date)
+
+    return procedure_list
+
+
+def get_diagnosis_list(length):
+    diagnosis_list = []
+    for i in range(2, 26):
+        diagnosis = {'fieldid': f'odiag{i}', 'name': f'Diagnosis {i}', 'length': length, 'justification': 'left'}
+        diagnosis_list.append(diagnosis)
+        present_on_admission = {'fieldid': f'opoa{i}', 'name': f'Present on Admission {i}', 'length': 1, 'justification': 'left'}
+        diagnosis_list.append(present_on_admission)
+
+    return diagnosis_list
+
+
+def get_morbidity_list():
+    morbidity_list = []
+    for i in range(1, 13):
+        morbidity = {'fieldid': f'ecm{i}', 'name': f'External Causes of Morbidity {i}', 'length': 8, 'justification': 'left'}
+        morbidity_list.append(morbidity)
+        present_on_admission = {'fieldid': f'epoa{i}', 'name': f'POA - External Causes of Morbidity {i}', 'length': 1,
+                                'justification': 'left'}
+        morbidity_list.append(present_on_admission)
+
+    return morbidity_list
+
+
+def modify_row(row, df_fields, new_fields):
```

Comment:
This probably needs a more descriptive name, like `add_fields_split_by_index` or something.